inc/posix/private/getopt.h

```diff
@@ -1,8 +1,18 @@
 #ifndef _PRIVATE_GETOPT_H
 #define _PRIVATE_GETOPT_H
 
+struct z_getopt_r_context {
+    char *optarg;
+    int opterr;
+    int optind;
+    int optopt;
+    int optcur;
+};
+
+int z_getopt_r(struct z_getopt_r_context *context, int argc, char *const argv[], const char *optstring);
+
 extern char *optarg;
 extern int opterr, optind, optopt;
-int getopt(int, char * const [], const char *);
+int getopt(int argc, char *const argv[], const char *optstring);
 
 #endif /* _PRIVATE_GETOPT_H */
```

Comment:
these seems like private implementation details, I don't think we want or need any of these changes in the public libc headers

Reply:
`optcur` is an implementation detail, while `optarg`, `opterr`, `optind`, `optopt` behave as specified. if you have a suggestion to hide this implementation detail, feel free to tell me :)

parameter names are useful for auto-completion and the like.

Reply:
parameter names seem fine, why are we exposing the `z_getopt_r_context` and `z_getopt_r` symbols?  Are these part of libc?  If not, why are they in the libc headers? Why are we exporting them?